File: LayarTancep/app/src/main/java/com/dicoding/cinemaxx/MainActivity.java
package com.dicoding.cinemaxx;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.content.res.TypedArray;
import android.os.Bundle;
import android.view.View;
import android.widget.AdapterView;
import android.widget.AdapterView.OnItemClickListener;
import android.widget.ListView;
import android.widget.Toast;

import java.util.ArrayList;

import static android.os.Looper.prepare;

public class MainActivity extends AppCompatActivity {
    private String[] filmName;
    private String[] filmDesc;
    private String[] filmGenre;
    private String[] filmArtist;
    private TypedArray imgFilm;
    private  FilmAdapter adapter;
    private ArrayList<Film> films;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        adapter = new FilmAdapter(this);

        ListView listView = findViewById(R.id.lv_list);
        listView.setAdapter(adapter);

        prepare();
        addItem();

        listView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> adapterView, View view, int i, long l) {
                Toast.makeText(MainActivity.this, films.get(i).getName(), Toast.LENGTH_SHORT).show();

                Intent detailFilm = new Intent(MainActivity.this, DetailFilmActivity.class);
                detailFilm.putExtra(DetailFilmActivity.EXTRA_FILM, films.get(i));
                startActivity(detailFilm);
            }
        });
    }

    private void prepare(){
        filmName = getResources().getStringArray(R.array.film_title);
        filmDesc = getResources().getStringArray((R.array.film_description));
        filmGenre = getResources().getStringArray(R.array.film_genre);
        filmArtist = getResources().getStringArray(R.array.artist);
        imgFilm = getResources().obtainTypedArray(R.array.film_photo);
    }

    private void addItem(){
        films = new ArrayList<>();

        for (int i = 0; i < filmName.length; i++){
            Film film = new Film();
            film.setPhoto(imgFilm.getResourceId(i, -1));
            film.setName(filmName[i]);
            film.setDescription(filmDesc[i]);
            film.setArtist(filmArtist[i]);
            film.setGenre(filmGenre[i]);
            films.add(film);
        }

        adapter.setFilms(films);
    }


}

File: MyListView/app/src/main/java/com/xantrikoding/mylistview/MainActivity.kt
package com.xantrikoding.mylistview

import android.content.res.TypedArray
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.ArrayAdapter
import android.widget.BaseAdapter
import android.widget.*
import com.xantrikoding.mylistview.adapter.HeroAdapter
import com.xantrikoding.mylistview.databinding.ActivityMainBinding
import com.xantrikoding.mylistview.model.Hero

class MainActivity : AppCompatActivity() {
    private lateinit var adapter: HeroAdapter
    private var heroes = arrayListOf<Hero>()

    private lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        binding.lvList

        adapter = HeroAdapter(this)

        binding.lvList.adapter = adapter

// fungsi prepare untuk inisialisasi array
        prepare()

// setelah datanya disiapkan pada fungsi prepare(), lanjut membuat fungsi baru


// setelah data disiapkan dan ada fungsi untuk memasukkan data, lanjut membuat metode setOnItem
        binding.lvList.onItemClickListener = AdapterView.OnItemClickListener { _, _, position, _ ->
            Toast.makeText(this@MainActivity, heroes[position].name, Toast.LENGTH_SHORT).show()
        }
    }

    private fun prepare() {
        val dataName = resources.getStringArray(R.array.data_name)
        val dataDescription = resources.getStringArray(R.array.data_description)
        val dataPhoto = resources.obtainTypedArray(R.array.data_photo)

        val  heroes = ArrayList<Hero> ()
        for (position in dataName.indices) {
            val hero = Hero(
                dataPhoto.getResourceId(position, -1),
                dataName[position],
                dataDescription[position]
            )
            heroes.add(hero)
        }
        adapter.heroes = heroes
    }
}

/** baris kode menggunakan binding
private lateinit var adapter: HeroAdapter
private var heroes = arrayListOf<Hero>()

private lateinit var binding: ActivityMainBinding

override fun onCreate(savedInstanceState: Bundle?) {
super.onCreate(savedInstanceState)
binding = ActivityMainBinding.inflate(layoutInflater)
setContentView(binding.root)

binding.lvList

// kalau pakai val di adapter maka itu akan membentuk variabel baru, padahal adapter disini memanggil adapter yg di atas
adapter = HeroAdapter(this)
/**
 * baris kode di bawah tidak digunakan lagi karena variabel di atas diganti
ArrayAdapter<String> (this,
// kombinasi simpe_list_item1 dan text 2 membuat app Force Close
android.R.layout.simple_list_item_1, android.R.id.text1, dataName)
*/
binding.lvList.adapter = adapter

// fungsi prepare untuk inisialisasi array
prepare()

// setelah datanya disiapkan pada fungsi prepare(), lanjut membuat fungsi baru


// setelah data disiapkan dan ada fungsi untuk memasukkan data, lanjut membuat metode setOnItem
binding.lvList.onItemClickListener = AdapterView.OnItemClickListener { _, _, position, _ ->
Toast.makeText(this@MainActivity, heroes[position].name, Toast.LENGTH_SHORT).show()
}
}

private fun prepare() {
val dataName = resources.getStringArray(R.array.data_name)
val dataDescription = resources.getStringArray(R.array.data_description)
val dataPhoto = resources.obtainTypedArray(R.array.data_photo)

val  heroes = ArrayList<Hero> ()
for (position in dataName.indices) {
val hero = Hero(
dataPhoto.getResourceId(position, -1),
dataName[position],
dataDescription[position]
)
heroes.add(hero)
}
adapter.heroes = heroes
}
 */

/*
private lateinit var adapter: HeroAdapter
    private lateinit var dataName: Array<String>
    private lateinit var dataDescription: Array<String>
    private lateinit var dataPhoto: TypedArray
    private var heroes = arrayListOf<Hero>()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val listView: ListView = findViewById(R.id.lv_list)
        // kalau pakai val di adapter maka itu akan membentuk variabel baru, padahal adapter disini memanggil adapter yg di atas
        adapter = HeroAdapter(this)
        listView.adapter = adapter

        // fungsi prepare untuk inisialisasi array
        prepare()

        // setelah datanya disiapkan pada fungsi prepare(), lanjut membuat fungsi baru
        addItem()

        // setelah data disiapkan dan ada fungsi untuk memasukkan data, lanjut membuat metode setOnItem
        listView.onItemClickListener = AdapterView.OnItemClickListener { _, _, position, _ ->
            Toast.makeText(this@MainActivity, heroes[position].name, Toast.LENGTH_SHORT).show()
        }
    }

    private fun addItem() {
        for (position in dataName.indices) {
            val hero = Hero(
                dataPhoto.getResourceId(position, -1),
                dataName[position],
                dataDescription[position]
            )
            heroes.add(hero)
        }
        adapter.heroes = heroes
    }

    private fun prepare() {
        dataName = resources.getStringArray(R.array.data_name)
        dataDescription = resources.getStringArray(R.array.data_description)
        dataPhoto = resources.obtainTypedArray(R.array.data_photo)
    }
 */